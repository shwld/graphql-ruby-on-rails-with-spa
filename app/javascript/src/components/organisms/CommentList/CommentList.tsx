import React, { useEffect, useState } from 'react'

import { useOnMessageAddedSubscription } from '@/graphql/generated'

export const CommentListTemplate: React.VFC<{
  messages: string[]
  className?: string
}> = ({ messages, className = '' }) => {
  return (
    <div className={className}>
      {messages.map((message, i) => (
        <p key={i}>{message}</p>
      ))}
    </div>
  )
}

export const CommentList: React.VFC<{ className?: string }> = ({
  className,
}) => {
  const [messages, setMessages] = useState<string[]>([])
  const messageAdded = useOnMessageAddedSubscription()
  useEffect(() => {
    const message = messageAdded.data?.onMessageAdded.message
    if (!message) return
    setMessages((oldMessages) => [...oldMessages, message])
  }, [messageAdded.data?.onMessageAdded, setMessages])

  return <CommentListTemplate messages={messages} className={className} />
}
