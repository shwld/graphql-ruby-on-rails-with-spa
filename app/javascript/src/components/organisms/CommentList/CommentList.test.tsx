import React from 'react'
import { create } from 'react-test-renderer'

import { CommentListTemplate } from './CommentList'

describe('Organisms/CommentList', () => {
  it('Snap Shot', () => {
    const component = create(<CommentListTemplate messages={['message']} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
