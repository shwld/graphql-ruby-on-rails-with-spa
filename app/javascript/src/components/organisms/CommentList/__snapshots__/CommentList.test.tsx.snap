// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organisms/CommentList Snap Shot 1`] = `
<div
  className=""
>
  <p>
    message
  </p>
</div>
`;
